python/cuml/dask/common/base.py

```diff
@@ -58,41 +145,42 @@ def __getattr__(self, attr):
         If the attribute being requested is not directly on the local object,
         this function will see if the local object contains the attribute
         prefixed with an _. In the case the attribute does not exist on this
-        local instance, the request will be proxied to self.local_model and
+        local instance, the request will be proxied to self.internal_model and
         will be fetched either locally or remotely depending on whether
-        self.local_model is a local object instance or a future.
+        self.internal_model is a local object instance or a future.
         """
         real_name = '_' + attr
 
-        # First check locally for attr
-        if attr in self.__dict__:
-            ret_attr = self.__dict__[attr]
+        internal_model = self._get_internal_model()
```

Comment:
My initial not very good idea - allow a model to optionally have a "_model_attributes_" list that would say what attrs are on the remote model. We'd only use this in the case where it might be expensive to grab the internal model (e.g. rf)

Reply:
Another possible idea is just to move the get_internal_model later so that it's only called if other ways of finding attrs fail?

Reply:
Actually, the doc of `get_internal_model` needed to be changed. Originally, I was thinking the combined model would be created using `get_internal_model` but it seemed weird to have RF combine the model in a getter so I changed it to happen in the setter. I'll update the doc to `_get_internal_model()` to make sure nothing happens in the setter but I'll make this change as well just in case a user doesn't see it.